src/toil/jobStores/googleJobStore.py

```diff
@@ -158,17 +158,26 @@ def destroy(self):
         except exceptions.NotFound:
             # just return if not connect to physical storage. Needed for idempotency
             return
+        
         try:
             self.bucket.delete(force=True)
-            # throws ValueError if bucket has more than 256 objects. Then we must delete manually
         except ValueError:
-            self.bucket.delete_blobs(self.bucket.list_blobs())
+            with self.storageClient.batch():
+                for blob in self.bucket.list_blobs():
```

Comment:
It looks like there might be [a limit of 1000 total items](https://medium.com/analytics-vidhya/pagination-batching-restricting-result-data-with-google-cloud-storage-python-client-library-b87bc50393d) on the size of a batch. If there are more than 1000 items to delete, this might not work.

I did some research and it looks like [`delete_blobs` is documented to just call `delete_blob`, with none of its own batching](https://googleapis.dev/python/storage/latest/buckets.html?highlight=delete_blobs#google.cloud.storage.bucket.Bucket.delete_blobs).

I think overall what you should probably do is break the list of blobs up into groups of `MAX_BATCH_SIZE` or less, and for each group, make a  `self.storageClient.batch()` and insidfe it a single call to `self.storageClient.delete_blobs()` with all the blobs in the group. Then you don't need the inner loop over individual blobs.

The most important thing, though, is to explain in comments why you've decided to do it the way you're doing it. Otherwise someone is going to "simplify" away all the work you did to make it fast, and replace it with a single `delete_blobs()` again.